airflow/utils/timezone.py

```diff
@@ -179,8 +179,7 @@ def coerce_datetime(v: Union[None, dt.datetime, DateTime]) -> Optional[DateTime]
     """Convert whatever is passed in to an timezone-aware ``pendulum.DateTime``."""
     if v is None:
         return None
-    if v.tzinfo is None:
-        v = make_aware(v)
     if isinstance(v, DateTime):
-        return v
-    return pendulum.instance(v)
+        return v if v.tzinfo else make_aware(v)
```

Comment:
How well covered was this function by tests? I'm not 100% sure this has the same behaviour

Reply:
Very good point. I've added tests and run it  before/after and behaviour is the same :)